### PR TITLE
React to Logging API changes

### PR DESCRIPTION
**Pull request by BrennanConroy**

### CODE REVIEW

File: src/Microsoft.AspNetCore.AzureAppServicesIntegration/AppServicesWebHostBuilderExtensions.cs [hunk 1]
```diff
@@ -20,7 +20,7 @@ public static IWebHostBuilder UseAzureAppServices(this IWebHostBuilder hostBuild
                 throw new ArgumentNullException(nameof(hostBuilder));
             }
 
-            hostBuilder.ConfigureLogging(loggerFactory => loggerFactory.AddAzureWebAppDiagnostics());
+            hostBuilder.ConfigureLogging<LoggerFactory>((_, loggerFactory) => loggerFactory.AddAzureWebAppDiagnostics());
```
Comment (BrennanConroy) [inline src/Microsoft.AspNetCore.AzureAppServicesIntegration/AppServicesWebHostBuilderExtensions.cs]: 🆙 📅 Changed this too

File: test/Microsoft.AspNetCore.AzureAppServicesIntegration.Tests/AppServicesWebHostBuilderExtensionsTest.cs [hunk 2]
```diff
@@ -17,7 +17,7 @@ public void UseAzureAppServices_RegisterLogger()
 
             mock.Object.UseAzureAppServices();
 
-            mock.Verify(builder => builder.ConfigureLogging(It.IsNotNull<Action<ILoggerFactory>>()), Times.Once);
+            mock.Verify(builder => builder.ConfigureLogging((It.IsNotNull<Action<WebHostBuilderContext, LoggerFactory>>())), Times.Once);
```
Comment (Eilon) [inline test/Microsoft.AspNetCore.AzureAppServicesIntegration.Tests/AppServicesWebHostBuilderExtensionsTest.cs]: Are the extra parens needed here?

File: sample/ApplicationInsightsHostingStartupSample/Startup.cs [hunk 3]
```diff
@@ -50,6 +51,9 @@ public void Configure(IApplicationBuilder app, ILoggerFactory loggerFactory, ISe
                 specificLogger.LogInformation("Specific information log");
                 specificLogger.LogWarning("Specific warning log");
 
+                // Allow time for logger to log
+                await Task.Delay(100);
```
Comment (pakrym) [inline sample/ApplicationInsightsHostingStartupSample/Startup.cs]: Does AI dispatch?

File: sample/ApplicationInsightsHostingStartupSample/Startup.cs [hunk 4]
```diff
@@ -107,6 +108,11 @@ public static void Main(string[] args)
                 .Build();
 
             var host = new WebHostBuilder()
+                .ConfigureLogging((hostingContext, factory) =>
+                {
+                    factory.UseConfiguration(hostingContext.Configuration.GetSection("Logging"))
```
Comment (BrennanConroy) [inline sample/ApplicationInsightsHostingStartupSample/Startup.cs]: @mikeharder This solves the test failure
@pakrym Added this after your review

File: sample/ApplicationInsightsHostingStartupSample/Startup.cs [hunk 5]
```diff
@@ -23,12 +24,10 @@ public void ConfigureServices(IServiceCollection services)
 
         public void Configure(IApplicationBuilder app, ILoggerFactory loggerFactory, IServiceProvider serviceProvider)
         {
-            loggerFactory.AddConsole(LogLevel.Debug);
-
             var logger = loggerFactory.CreateLogger("Requests");
 
             app.UseMvcWithDefaultRoute();
-            app.Map("/log", logApp => logApp.Run(async (context) =>
+            app.Map("/log", logApp => logApp.Run((context) =>
```
Comment (Eilon) [inline sample/ApplicationInsightsHostingStartupSample/Startup.cs]: BTW can remove the parens around `(context)` now.

**Pull request merged**